src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -461,6 +461,8 @@ public virtual bool IsImplicitClass
             }
         }
 
+        internal virtual bool IsRecord => false;
```

Comment:
>internal virtual bool IsRecord => false; [](start = 8, length = 40)

It looks like we are missing some overrides. For a long time we are following a rule. No new virtual APIs should be added to symbols. Only non-virtual or abstract ones. Only leaf classes can implement the abstract APIs